src/main/java/seedu/bankwithus/Parser.java

```diff
@@ -107,22 +118,24 @@ public void parseUserInput(String input) throws CommandNotFoundException, IOExce
      * @param scanner
      * @throws CorruptedSaveFileException if any of the parameters are corrupted
      */
-    public void parseSavedFile(Scanner scanner) throws CorruptedSaveFileException, SaveFileIsEmpty {
-        String accountDetails = scanner.nextLine();
-        if (accountDetails.isBlank()) {
-            throw new SaveFileIsEmpty();
-        }
+    public void parseSavedFile(Scanner scanner) throws CorruptedSaveFileException, SaveFileIsEmptyException {
         try {
-            String[] splitDetails = accountDetails.split(";");
-            String name = splitDetails[0].trim();
-            String balanceString = splitDetails[1].trim();
-            if (name.isEmpty() || balanceString.isEmpty()) {
-                throw new Exception();
+            while (scanner.hasNextLine()) {
+                String accountDetails = scanner.nextLine();
+                if (accountDetails.isBlank()) {
+                    throw new SaveFileIsEmptyException();
+                }
+                String[] splitDetails = accountDetails.split(";");
+                String name = splitDetails[0].trim();
+                String balanceString = splitDetails[1].trim();
+                if (name.isEmpty() || balanceString.isEmpty()) {
+                    throw new Exception();
```

Comment:
Throw CorruptedSavefileException to be safe. I get the feeling that this will be caught in the first catch block, which is usually not what we want

Reply:
Cancel last, just realised that it doesnt matter cos everything will throw runtime exception anyway